### DISCUSSION TITLE
transparent proxy run failed

### DISCUSSION THREAD

**Discussion by stark-ydq** [flomesh-io/pipy#191] [Q&A]
hello when I run pipy/samples/bpf/transparent-proxy ,here is an error
"2024-07-16 15:58:22.344 [ERR] [pjs] File /main.js:
2024-07-16 15:58:22.345 [ERR] [pjs] Line 5:  var progCgConnect4 = obj.programs.find(p => p.name === 'cg_connect4').load('BPF_PROG_TYPE_CGROUP_SOCK_ADDR', 'BPF_CGROUP_INET4_CONNECT')
2024-07-16 15:58:22.345 [ERR] [pjs]                                                                                    ^
2024-07-16 15:58:22.345 [ERR] [pjs] Error: syscall BPF_MAP_CREATE failed with errno = 22: Invalid argument
2024-07-16 15:58:22.345 [ERR] [pjs] Backtrace:" could you please help me

**Comment by ethinx** [comment]
have you compiled the bpf module `transparent-proxy.o` in advanced? 

as the ebpf function depends on the kernel, could you provide some details about the running environment?

OS version:
kernel version:

**Reply by stark-ydq** [reply]
OS version:NAME="Tencent tlinux"
VERSION="2.6"
ID="tlinux"
ID_LIKE="rhel fedora centos"
VERSION_ID="2.6"
PRETTY_NAME="Tencent tlinux 2.6"

kernel version:4.14.105-1-tlinux3-0023.1  ,And I have already compiled the bpf module transparent-proxy.o in advanced,no error,generated transparent.o

**Reply by ethinx** [reply]
The kernel may be a little old to run the bpf module. If possible, you could try on some newer version  OS, like ubuntu.

The type of bpf program we used and the corresponding kernel version as following:

BPF_PROG_TYPE_CGROUP_SOCK_ADDR - 4.17
BPF_PROG_TYPE_SOCK_OPS - 4.13
BPF_PROG_TYPE_CGROUP_SOCKOPT - 5.3

details could be check on https://github.com/iovisor/bcc/blob/master/docs/kernel-versions.md

and you could check the file `/usr/include/linux/bpf.h` on you OS, the `bpf_attach_type` enum. It would list the supported BPF hooks.

We haven't tried on Tencent tlinux yet, and may need some time to investigate if they are compatible.
